src/hooks/useFetchExperience.ts

```diff
@@ -0,0 +1,173 @@
+import type { ContentfulClientApi, Entry } from 'contentful'
+import { useCallback, useState } from 'react'
+import { fetchExperienceEntities, fetchExperienceEntry } from '../core/fetchers'
+import { Experience, ExternalSDKMode } from '../types'
+import { createExperience } from '../core'
+
+const errorMessagesWhileFetching = {
+  experience: 'Failed to fetch experience',
+  experienceReferences: 'Failed to fetch entities, referenced in experience',
+}
+
+const handleError = (generalMessage: string, error: unknown) => {
+  const message = error instanceof Error ? error.message : `Unknown error: ${error}`
+  console.error(`${generalMessage} with error: ${message}`)
+}
+
+type useClientsideExperienceFetchersProps = {
+  mode: ExternalSDKMode
+  client: ContentfulClientApi<undefined>
+}
+
+export const useFetchExperience = ({ mode, client }: useClientsideExperienceFetchersProps) => {
+  const [experience, setExperience] = useState<Experience | undefined>(undefined)
+  const [isFetching, setIsFetching] = useState(false)
+
+  /**
+   * Fetch experience entry using slug as the identifier
+   * @param {string} experienceTypeId - id of the content type associated with the experience
+   * @param {string} slug - slug of the experience (defined in entry settings)
+   * @param {string} localeCode - locale code to fetch the experience. Falls back to the currently active locale in the state
+   */
+  const fetchBySlug = useCallback(
+    async ({
+      experienceTypeId,
+      slug,
+      localeCode,
+    }: {
+      experienceTypeId: string
+      slug: string
+      localeCode: string
+    }): Promise<Experience | undefined> => {
+      setIsFetching(true)
```

Comment:
One thing that I learned today can save a few lines in this method again:
```
setIsFetching(true);
try {
  // do anything else and even return things
} finally {
  // ensure that the boolean gets reset before the function exits
  setIsFetching(false);
}
```

Reply:
See my comment below how this helps making the function way shorter :)

Reply:
It used to work like this, but I wanted to have granularity in terms of what we print to the console. For example, when the function fetches the entry, I wanted to print to the console the error which says "Failed to fetch the experience entry", but for example, when the function fetches the references in the entry, I wanted to print "Failed to fetch the experience entities" or similar.

One try catch resulted in a generic print log, which of course could be resolved if I created different errors types, but this PR already seemed quite big even without it

Reply:
You can still have multiple try/catch blockes within the try/finally block but I agree that this wouldn't be an actual code improvement :)
I addressed this in the comment below, so let me know if that works for you :)